program/rust/src/processor/add_publisher.rs

```diff
@@ -73,13 +73,21 @@ pub fn add_publisher(
         }
     }
 
-    let current_index: usize = try_convert(price_data.num_)?;
+    let mut current_index: usize = try_convert(price_data.num_)?;
     sol_memset(
         bytes_of_mut(&mut price_data.comp_[current_index]),
         0,
         size_of::<PriceComponent>(),
     );
     price_data.comp_[current_index].pub_ = cmd_args.publisher;
+
+    while current_index > 0
```

Comment:
Would suggest adding a comment, PR title is clear but good to explain logic like this in place as well.